source/models/table.py

```diff
@@ -36,26 +37,69 @@ def get_path(self):
         
     def get_name(self):
         return self.__table_name
-
+    
     def set(self, row):
+        self.__colomns_name_validate(row = row)
         row_obj = Row(table = self, value = row)
         if row_obj.row_exists() and self.table_metadata.enable_overwrite == "false":
             raise RowExists(message = "this row is already exists and can't overwrite")
         elif row_obj.row_exists():
-            old_row = Row.load_by_primary_key(table = self, primary_key = row_obj.get_primary_key())
+            old_row = Row(table = self).load_by_primary_key(table=self, primary_key = row_obj.primary_key)
             old_row.delete()
         row_obj.serialize()
 
-    
-    def get(self):
-        pass
-    
+ 
+    def get_similar_primarykeys(self, query):
+        primarykeys = []
+        for index_name in query:
+            if index_name in self.table_metadata.index_keys:
+                primarykeys = self.table_metadata.indicies[index_name].get_primary_key(index_value = query[index_name]) 
+                break
+        return primarykeys
+ 
+    def get(self, query):
+        self.__colomns_name_validate(query)
+        primarykeys = []
+        if self.table_metadata.primary_key in query : # if primarykey in the query
+            primarykeys.append(query[self.table_metadata.primary_key])     
+ 
+        else:
+            primarykeys = self.get_similar_primarykeys(query = query)
+            if not len(primarykeys):# compare all data in the table with the query
+                primarykeys = self.get_all_primary_keys()
+        row_objects = self.get_rows(primarykeys)
+        return self.filter_by_query(row_objects = row_objects, query = query)
+
+    def get_rows(self, primarykeys):
+            row_objects = []
+            for primarykey in primarykeys:
+               row_objects.append(Row(self).load_by_primary_key(self, primarykey))
+            return row_objects
+
+    def filter_by_query(self, row_objects, query):
```

Comment:
Why this is an instance method ?  it doesn't use any attribute in table object, right ?